### DISCUSSION TITLE
Does jsonschema-generator work with any annotation libraries that auto-generate getters?

### DISCUSSION THREAD

**Discussion by kanesee** [victools/jsonschema-generator#322] [Q&A] [answered]
Wondering if it works with Lombok or Immutables or other libraries that auto-generate getters/setters?

I couldn't get it to work quite the way I want with Immutables because I have complicated inheritance in the mix.
Just to add, one of the issues I encountered trying to integrate Immutables with jsonschema-generator is that it doesn't respect `@JsonIgnore`, probably because the annotation is on my interface/abstract class but not on the generated immutable class.

e.g.
```
@Value.Immutable
@JsonDeserialize(as = ImmutableMyPojo.class)
@JsonSerialize(as = ImmutableMyPojo.class)
public abstract class MyPojo {
    @JsonIgnore
    public abstract MyEnum getMyEnum();
}
```

I haven't tried Lombok yet. It's a bit of a pain to introduce it into my company so just wanted to know what approach others have been success with first.

References to code examples would be awesome!

Thanks

**Comment by CarstenWickner** [comment, marked as answer]
Hi @kanesee,

Lombok and similar libraries are adding getters to the compiled Java byte code. The `jsonschema-generator` is looking at that byte code, i.e., those auto-generated getters should be treated like any old hand-crafted ones. At least that was my assumption so far.

The jackson `@JsonIgnore` annotation is being respected, if you include the `jsonschema-module-jackson` dependency and its corresponding configuration: https://victools.github.io/jsonschema-generator/#jackson-module
```java
configBuilder.with(new JacksonModule());
```

The challenge in your case appears to be that you expect the `@JsonSerialize` annotation to provide a drop-in replacement.
While that is not part of the default behavior, you could achieve it in multiple ways though.
One of them are "Target Type Overrides" (https://victools.github.io/jsonschema-generator/#target-type-overrides) such as this one:
```java
configBuilder.forFields().withTargetTypeOverridesResolver(field -> Optional
            .ofNullable(field.getAnnotationConsideringFieldAndGetterIfSupported(JsonSerialize.class))
            .map(JsonSerialize::as)
            .map(field.getContext()::resolve)
            .map(resolvedSubtype -> List.of(resolvedSubtype))
            .orElse(null));
```
